iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientConfiguration.java

```diff
@@ -91,6 +91,8 @@ public final class ClientConfiguration
 
     private boolean useIdentifiableThreadNames = true;
 
+    private boolean logRoutineDisconnectsAsErrors = true;
```

Comment:
It is a nice option, but maybe add an option in `ModuleClient.createFromEnvironment` to avoid to manipulate `ClientOptions` is a better approach. Something like ClientOptionsConfigurer interface to only manipulate the option you need to modify.

Reply:
```ModuleClient.createFromEnvironment``` already allows you to pass in a ```ClientOptions``` instance. If you only want to manipulate this new option, then it can already be done:

```java
ClientOptions options = ClientOptions.builder()
        .logRoutineDisconnectsAsErrors(false)
        .build();

ModuleClient.createFromEnvironment(unixDomainSocket, protocol, options);
```

Reply:
I don't follow. What is stopping you from using ```ModuleClient.createFromEnvironment(unixDomainSocket, protocol, options``` ?

Reply:
For sure, we can use `ModuleClient.createFromEnvironment(unixDomainSocket, protocol, options)` but the use of the default `ClientOptions` should also be acceptable.

In the case you are proposing, you have to start to care about the construction of `ClientOptions` and if you use `ModuleClient.createFromEnvironment(unixDomainSocket, protocol)` you don't have to do it. Maybe it doesn't have sense to configure `ClientOptions` when you only need to configure logs.

Reply:
I'm not going to make opting out of these logs as warnings/errors the default behavior, so you will need to start constructing a ```ClientOptions``` here. By changing the default behavior, we risk causing unexpected problems for existing users. The best I can do here is allow you to opt-in via ```ClientOptions```